### PR TITLE
Ansible tutorial TODOs from Barcelona (Helena edition)

### PR DESCRIPTION
**Pull request by hexylena**
xref https://github.com/galaxyproject/training-material/issues/1809

### CODE REVIEW

**Comment by hexylena** [comment]
@natefoo can you review?

File: topics/admin/tutorials/connect-to-compute-cluster/slides.html [hunk 1]
```diff
@@ -293,131 +293,3 @@
...
-Specifies the order in which to attempt to resolve tool dependencies, e.g.:
-```xml
-<dependency_resolvers>
-    <tool_shed_packages />
-    <galaxy_packages />
-    <conda />
-    <galaxy_packages versionless="true" />
-    <conda versionless="true" />
-    <modules modulecmd="/opt/Modules/3.2.9/bin/modulecmd" />
-    <modules modulecmd="/opt/Modules/3.2.9/bin/modulecmd" versionless="true" default_indicator="default" />
```
Comment (natefoo) [inline topics/admin/tutorials/connect-to-compute-cluster/slides.html]: 1 slide saying "Galaxy can use Modules, consult whatever slides the dependency resolver config ends up in" might be reasonable?

File: topics/admin/tutorials/connect-to-compute-cluster/tutorial.md [hunk 2]
```diff
@@ -71,30 +71,32 @@ be taken into consideration when choosing where to run jobs and what parameters
...
-> 2. In the same directory, run `ansible-galaxy install -p roles -r requirements.yml`. This will install all of the required modules for this training into the `roles/` folder. We choose to install to a folder to give you easy access to look through the different roles when you have questions on their behaviour.
+> 2. In the same directory, run:
 >
-> 3. Create the hosts file if you have not done so, include a group for `[galaxyservers]` with the address of the host where you will install Slurm
+>    ```
+>    ansible-galaxy install -p roles -r requirements.yml
+>    ```
 >
-> 4. Create a playbook, `slurm.yml` which looks like the following:
+> 4. Add the following roles to the *beginning* of your roles section in your `galaxy.yml` playbook:
```
Comment (natefoo) [inline topics/admin/tutorials/connect-to-compute-cluster/tutorial.md]:
```suggestion
> 3. Add the following roles to the *beginning* of your roles section in your `galaxy.yml` playbook:
```

File: topics/admin/tutorials/connect-to-compute-cluster/tutorial.md [hunk 3]
```diff
@@ -71,30 +71,32 @@ be taken into consideration when choosing where to run jobs and what parameters
...
+>
+>    - `galaxyproject.repos`
+>    - `galaxyproject.slurm`
+>
+> 5. Add the slurm variables to your  `group_vars/galaxyservers.yml`:
```
Comment (natefoo) [inline topics/admin/tutorials/connect-to-compute-cluster/tutorial.md]:
```suggestion
> 4. Add the slurm variables to your  `group_vars/galaxyservers.yml`:
```

File: topics/admin/tutorials/connect-to-compute-cluster/tutorial.md [hunk 4]
```diff
@@ -299,9 +294,9 @@ At the top of the stack sits Galaxy. Galaxy must now be configured to use the cl
 >
 > 2. We need to modify `job_conf.xml` to instruct Galaxy on how to use a more advanced job submission setup. We will begin with a basic job conf:
 >
->    If the folder does not exist, create `files/galaxy/config` next to your `galaxy.yml` playbook (`mkdir -p files/galaxy/config/`).
+>    If the folder does not exist, create `templates/galaxy/config` next to your `galaxy.yml` playbook (`mkdir -p templates/galaxy/config/`).
 >
->    Create `files/galaxy/config/job_conf.xml` with the following contents:
+>    Create `templates/galaxy/config/job_conf.xml` with the following contents:
```
Comment (natefoo) [inline topics/admin/tutorials/connect-to-compute-cluster/tutorial.md]:
```suggestion
>    Create `templates/galaxy/config/job_conf.xml.j2` with the following contents:
```

File: topics/admin/tutorials/connect-to-compute-cluster/tutorial.md [hunk 5]
```diff
@@ -316,14 +311,14 @@ At the top of the stack sits Galaxy. Galaxy must now be configured to use the cl
 >
 >    > ### {% icon comment %} Note
 >    >
->    > Depending on the order in which you are completing this tutorial in relation to other tutorials, you may have already created the `job_conf.xml` file, as well as defined `galaxy_config_files` and set the `job_config_file` option in `galaxy_config` (step 4). If this is the case, be sure to **merge the changes in this section with your existing playbook**.
+>    > Depending on the order in which you are completing this tutorial in relation to other tutorials, you may have already created the `job_conf.xml` file, as well as defined `galaxy_config_templates` and set the `job_config_file` option in `galaxy_config` (step 4). If this is the case, be sure to **merge the changes in this section with your existing playbook**.
```
Comment (natefoo) [inline topics/admin/tutorials/connect-to-compute-cluster/tutorial.md]:
```suggestion
>    > Depending on the order in which you are completing this tutorial in relation to other tutorials, you may have already created the `job_conf.xml.j2` file, as well as defined `galaxy_config_templates` and set the `job_config_file` option in `galaxy_config` (step 4). If this is the case, be sure to **merge the changes in this section with your existing playbook**.
```

File: topics/admin/tutorials/connect-to-compute-cluster/tutorial.md [hunk 6]
```diff
@@ -316,14 +311,14 @@ At the top of the stack sits Galaxy. Galaxy must now be configured to use the cl
...
 >    {: .comment}
 >
 > 3. Next, we need to configure the Slurm job runner. First, we instruct Galaxy's job handlers to load the Slurm job runner plugin, and set the Slurm job submission parameters. A job runner plugin definition must have the `id`, `type`, and `load` attributes. Then we add a basic destination with no parameters, Galaxy will do the equivalent of submitting a job as `sbatch /path/to/job_script.sh`. Note that we also need to set a default destination now that more than one destination is defined. In a `<destination>` tag, the `id` attribute is a unique identifier for that destination and the `runner` attribute must match the `id` of a defined plugin:
 >
 >    ```diff
->    --- files/galaxy/config/job_conf.xml.old
->    +++ files/galaxy/config/job_conf.xml
+>    --- templates/galaxy/config/job_conf.xml.old
+>    +++ templates/galaxy/config/job_conf.xml
```
Comment (natefoo) [inline topics/admin/tutorials/connect-to-compute-cluster/tutorial.md]:
```suggestion
>    --- templates/galaxy/config/job_conf.xml.j2.old
>    +++ templates/galaxy/config/job_conf.xml.j2
```

File: topics/admin/tutorials/connect-to-compute-cluster/tutorial.md [hunk 7]
```diff
@@ -351,20 +346,20 @@ At the top of the stack sits Galaxy. Galaxy must now be configured to use the cl
...
->    And then deploy the new config file using the `galaxy_config_files` var in your group vars:
+>    And then deploy the new config file using the `galaxy_config_templates` var in your group vars:
 >
 >    {% raw %}
 >    ```yaml
->    galaxy_config_files:
+>    galaxy_config_templates:
 >      # ... possible existing config file definitions
->      - src: files/galaxy/config/job_conf.xml
+>      - src: templates/galaxy/config/job_conf.xml
```
Comment (natefoo) [inline topics/admin/tutorials/connect-to-compute-cluster/tutorial.md]:
```suggestion
>      - src: templates/galaxy/config/job_conf.xml.j2
```

File: topics/admin/tutorials/connect-to-compute-cluster/tutorial.md [hunk 8]
```diff
@@ -542,11 +537,11 @@ We want our tool to run with more than one core. To do this, we need to instruct
 
 > ### {% icon hands_on %} Hands-on: Allocating more resources
 >
-> 1. Edit your `files/galaxy/config/job_conf.xml` and add the following destination:
+> 1. Edit your `templates/galaxy/config/job_conf.xml` and add the following destination:
```
Comment (natefoo) [inline topics/admin/tutorials/connect-to-compute-cluster/tutorial.md]:
```suggestion
> 1. Edit your `templates/galaxy/config/job_conf.xml.j2` and add the following destination:
```

File: topics/admin/tutorials/connect-to-compute-cluster/tutorial.md [hunk 9]
```diff
@@ -542,11 +537,11 @@ We want our tool to run with more than one core. To do this, we need to instruct
...
 >
 >    ```xml
 >            <destination id="slurm-2c" runner="slurm">
->                <param id="nativeSpecification">--nodes=1 --ntasks=2</param>
+>                <param id="nativeSpecification">--nodes=1 --ntasks=1 --cpus-per-task=4</param>
```
Comment (natefoo) [inline topics/admin/tutorials/connect-to-compute-cluster/tutorial.md]:
```suggestion
>                <param id="nativeSpecification">--nodes=1 --ntasks=1 --cpus-per-task=2</param>
```

File: topics/admin/tutorials/connect-to-compute-cluster/tutorial.md [hunk 10]
```diff
@@ -731,7 +726,7 @@ Such form elements can be added to tools without modifying each tool's configura
 
 > ### {% icon hands_on %} Hands-on: Configuring a Resource Selector
 >
-> 1. Create and open `files/galaxy/config/job_resource_params_conf.xml`
+> 1. Create and open `templates/galaxy/config/job_resource_params_conf.xml`
```
Comment (natefoo) [inline topics/admin/tutorials/connect-to-compute-cluster/tutorial.md]:
```suggestion
> 1. Create and open `templates/galaxy/config/job_resource_params_conf.xml.j2`
```

File: topics/admin/tutorials/connect-to-compute-cluster/tutorial.md [hunk 11]
```diff
@@ -754,13 +749,13 @@ Such form elements can be added to tools without modifying each tool's configura
 >        galaxy:
 >          job_resource_params_file: {% raw %}"{{ galaxy_config_dir }}/job_resource_params_conf.xml"{% endraw %}
 >      ...
->      galaxy_config_files:
+>      galaxy_config_templates:
 >        ...
->        - src: files/galaxy/config/job_resource_params_conf.xml
->          dest: {% raw %}"{{ galaxy_config['galaxy']['job_resource_params_file'] }}"{% endraw %}
+>        - src: templates/galaxy/config/job_resource_params_conf.xml
```
Comment (natefoo) [inline topics/admin/tutorials/connect-to-compute-cluster/tutorial.md]:
```suggestion
>        - src: templates/galaxy/config/job_resource_params_conf.xml.j2
```

File: topics/admin/tutorials/connect-to-compute-cluster/tutorial.md [hunk 12]
```diff
@@ -754,13 +749,13 @@ Such form elements can be added to tools without modifying each tool's configura
...
+>          dest: {% raw %}"{{ galaxy_config.galaxy.job_resource_params_file }}"{% endraw %}
 >      ```
 >
-> 3. Next, we define a new section in `job_conf.xml`: `<resources>`. This groups together parameters that should appear together on a tool form. Add the following section to your `files/galaxy/config/job_conf.xml`:
+> 3. Next, we define a new section in `job_conf.xml`: `<resources>`. This groups together parameters that should appear together on a tool form. Add the following section to your `templates/galaxy/config/job_conf.xml`:
```
Comment (natefoo) [inline topics/admin/tutorials/connect-to-compute-cluster/tutorial.md]:
```suggestion
> 3. Next, we define a new section in `job_conf.xml`: `<resources>`. This groups together parameters that should appear together on a tool form. Add the following section to your `templates/galaxy/config/job_conf.xml.j2`:
```

File: topics/admin/tutorials/gxadmin/tutorial.md [hunk 13]
```diff
@@ -50,9 +50,18 @@ Since then it became the home for "all of the SQL queries we [galaxy admins] run
 >
 > 2. Install the role with `ansible-galaxy install -p roles -r requirements.yml`
 >
-> 3. Add the role to your playbook, it should run as root
+> 3. Add the following variables to your group variables file:
```
Comment (natefoo) [inline topics/admin/tutorials/gxadmin/tutorial.md]: Mention that maybe you already have them set if you ran the Grafana tutorial first?

File: topics/admin/tutorials/interactive-tools/tutorial.md [hunk 14]
```diff
@@ -457,9 +457,9 @@ A few Interactive Tool wrappers are provided with Galaxy, but they are [commente
 >
 > 1. Rather than modify the default tool configuration file, we'll add a new one that only references the Interactive Tools. This way, the default set of tools will still load without us having to incorporate the entire default tool config in to our playbook.
 >
->    If the folder does not exist, create `files/galaxy/config` next to your `galaxy.yml` (`mkdir -p files/galaxy/config/`)
+>    If the folder does not exist, create `templates/galaxy/config` next to your `galaxy.yml` (`mkdir -p templates/galaxy/config/`)
 >
->    Create `files/galaxy/config/tool_conf_interactive.xml` with the following contents:
+>    Create `templates/galaxy/config/tool_conf_interactive.xml` with the following contents:
```
Comment (natefoo) [inline topics/admin/tutorials/interactive-tools/tutorial.md]:
```suggestion
>    Create `templates/galaxy/config/tool_conf_interactive.xml.j2` with the following contents:
```

File: topics/admin/tutorials/interactive-tools/tutorial.md [hunk 15]
```diff
@@ -471,7 +471,7 @@ A few Interactive Tool wrappers are provided with Galaxy, but they are [commente
 >
 > 2. We need to modify `job_conf.xml` to instruct Galaxy on how run Interactive Tools (and specifically, how to run them in Docker). We will begin with a basic job conf:
 >
->    Create `files/galaxy/config/job_conf.xml` with the following contents:
+>    Create `templates/galaxy/config/job_conf.xml` with the following contents:
```
Comment (natefoo) [inline topics/admin/tutorials/interactive-tools/tutorial.md]:
```suggestion
>    Create `templates/galaxy/config/job_conf.xml.j2` with the following contents:
```

File: topics/admin/tutorials/interactive-tools/tutorial.md [hunk 16]
```diff
@@ -485,7 +485,7 @@ A few Interactive Tool wrappers are provided with Galaxy, but they are [commente
 >    ```
 >
 >    > ### {% icon comment %} Note
->    > Depending on the order in which you are completing this tutorial in relation to other tutorials, you may have already created the `job_conf.xml` file, as well as defined `galaxy_config_files` and set the `job_config_file` option in `galaxy_config` (step 4). If this is the case, be sure to **merge the changes in this section with your existing playbook**.
+>    > Depending on the order in which you are completing this tutorial in relation to other tutorials, you may have already created the `job_conf.xml` file, as well as defined `galaxy_config_templates` and set the `job_config_file` option in `galaxy_config` (step 4). If this is the case, be sure to **merge the changes in this section with your existing playbook**.
```
Comment (natefoo) [inline topics/admin/tutorials/interactive-tools/tutorial.md]:
```suggestion
>    > Depending on the order in which you are completing this tutorial in relation to other tutorials, you may have already created the `job_conf.xml.j2` file, as well as defined `galaxy_config_templates` and set the `job_config_file` option in `galaxy_config` (step 4). If this is the case, be sure to **merge the changes in this section with your existing playbook**.
```

File: topics/admin/tutorials/interactive-tools/tutorial.md [hunk 17]
```diff
@@ -521,7 +521,7 @@ A few Interactive Tool wrappers are provided with Galaxy, but they are [commente
 >
 >    {% raw %}
 >    ```yaml
->    galaxy_tool_config_files:
+>    galaxy_tool_config_templates:
```
Comment (natefoo) [inline topics/admin/tutorials/interactive-tools/tutorial.md]:
```suggestion
>    galaxy_tool_config_files:
```

This is a [role variable](https://github.com/galaxyproject/ansible-galaxy/blob/9b95b2b8ce34981649ceb475550683e95b181323/defaults/main.yml#L145)

File: topics/admin/tutorials/interactive-tools/tutorial.md [hunk 18]
```diff
@@ -537,20 +537,20 @@ A few Interactive Tool wrappers are provided with Galaxy, but they are [commente
...
->    And then deploy the new config files using the `galaxy_config_files` var in your group vars:
+>    And then deploy the new config templates using the `galaxy_config_templates` var in your group vars:
 >
 >    {% raw %}
 >    ```yaml
->    galaxy_config_files:
+>    galaxy_config_templates:
 >      # ... possible existing config file definitions
->      - src: files/galaxy/config/tool_conf_interactive.xml
+>      - src: templates/galaxy/config/tool_conf_interactive.xml
```
Comment (natefoo) [inline topics/admin/tutorials/interactive-tools/tutorial.md]:
```suggestion
>      - src: templates/galaxy/config/tool_conf_interactive.xml.j2
```

File: topics/admin/tutorials/interactive-tools/tutorial.md [hunk 19]
```diff
@@ -537,20 +537,20 @@ A few Interactive Tool wrappers are provided with Galaxy, but they are [commente
...
 >        dest: "{{ galaxy_config_dir }}/tool_conf_interactive.xml"
->      - src: files/galaxy/config/job_conf.xml
+>      - src: templates/galaxy/config/job_conf.xml
```
Comment (natefoo) [inline topics/admin/tutorials/interactive-tools/tutorial.md]:
```suggestion
>      - src: templates/galaxy/config/job_conf.xml.j2
```

File: topics/admin/tutorials/interactive-tools/tutorial.md [hunk 20]
```diff
@@ -664,7 +664,7 @@ Because we want to maintain dataset privacy, Pulsar is the better choice here. A
 >    ```
 >    {% endraw %}
 >
-> 2. Modify the job configuration file, `files/galaxy/config/job_conf.xml`, to configure Interactive Tools to use the embedded Pulsar runner.
+> 2. Modify the job configuration file, `templates/galaxy/config/job_conf.xml`, to configure Interactive Tools to use the embedded Pulsar runner.
```
Comment (natefoo) [inline topics/admin/tutorials/interactive-tools/tutorial.md]:
```suggestion
> 2. Modify the job configuration file, `templates/galaxy/config/job_conf.xml.j2`, to configure Interactive Tools to use the embedded Pulsar runner.
```

File: topics/admin/tutorials/job-metrics/tutorial.md [hunk 21]
```diff
@@ -53,7 +53,7 @@ These include very basic submission parameters. We want more information!
 
 > ### {% icon hands_on %} Hands-on: Setting up the job metrics file
 >
-> 1. Create the file `files/galaxy/config/job_metrics_conf.xml` with the following contents:
+> 1. Create the file `templates/galaxy/config/job_metrics_conf.xml` with the following contents:
```
Comment (natefoo) [inline topics/admin/tutorials/job-metrics/tutorial.md]:
```suggestion
> 1. Create the file `templates/galaxy/config/job_metrics_conf.xml.j2` with the following contents:
```

File: topics/admin/tutorials/job-metrics/tutorial.md [hunk 22]
```diff
@@ -92,14 +92,14 @@ These include very basic submission parameters. We want more information!
...
->      galaxy_config_files:
->    +   - src: files/galaxy/config/job_metrics_conf.xml
->    +     dest: "{{ galaxy_job_metrics_config_file }}"
->        - src: files/galaxy/config/tool_conf_interactive.xml
->          dest: "{{ galaxy_config_dir }}/tool_conf_interactive.xml"
->        - src: files/galaxy/config/job_conf.xml
+>     gie_proxy_sessions_path: "{{ galaxy_mutable_data_dir }}/interactivetools_map.sqlite"
+>
+>     galaxy_config_templates:
+>    +  - src: templates/galaxy/config/job_metrics_conf.xml
```
Comment (natefoo) [inline topics/admin/tutorials/job-metrics/tutorial.md]:
```suggestion
>    +  - src: templates/galaxy/config/job_metrics_conf.xml.j2
```

File: topics/admin/tutorials/object-store/slides.html [hunk 23]
```diff
@@ -15,211 +15,9 @@
...
-  * Works well in combination with `ftp_upload_dir`.
-* `allow_path_paste`
-  * Admin-only, allows importing from any path that the Galaxy's user has access to.
-
-???
-If you use old library interface you can also set `library_import_dir`.
-It specifies which folder admins may browse and import from.
 
 ---
-# Libraries Exercise
```
Comment (natefoo) [inline topics/admin/tutorials/object-store/slides.html]: Did we capture this somewhere?

File: topics/admin/tutorials/connect-to-compute-cluster/slides.html (hunk 1 above)
Comment (hexylena) [reply topics/admin/tutorials/connect-to-compute-cluster/slides.html]: yeah good idea!

File: topics/admin/tutorials/interactive-tools/tutorial.md (hunk 17 above)
Comment (hexylena) [reply topics/admin/tutorials/interactive-tools/tutorial.md]: too much regexing. thanks

File: topics/admin/tutorials/job-metrics/tutorial.md [hunk 24]
```diff
@@ -92,14 +92,14 @@ These include very basic submission parameters. We want more information!
...
+>    +    dest: "{{ galaxy_job_metrics_config_file }}"
```
[the remark this answers is not in the document]
Comment (hexylena) [reply topics/admin/tutorials/job-metrics/tutorial.md]: I guess we should decide. @shiltemann @mvdbeek @nsoranzo what do we think? Is it better for students (easier to understand, whatever else "better for them" means):

```
some_var: "{{ galaxy_dir }}/path"

galaxy_config:
  galaxy:
    job_path: "{{ some_var }}"

templates:
- src: something fun
  dest: "{{ some_var }}"
```

OR

```
galaxy_config:
  galaxy:
    job_path: "{{ galaxy_dir }}/path"

templates:
- src: something fun
  dest: "{{ galaxy_config.galaxy.job_path }}"
```

nate and I both do the second, but we've gone back and forth in teaching.

File: topics/admin/tutorials/object-store/slides.html (hunk 23 above)
Comment (hexylena) [reply topics/admin/tutorials/object-store/slides.html]: Not yet. It's in history, but this wasn't an execrise anyone was running anyway. I'm sketching up  new tutorial.

Comment (nsoranzo) [reply topics/admin/tutorials/object-store/slides.html]: @natefoo These libraries slides have been moved to topics/admin/tutorials/stuff/slides.html for the moment.

File: topics/admin/tutorials/job-metrics/tutorial.md (hunk 24 above)
[the remark this answers is not in the document]
Comment (nsoranzo) [reply topics/admin/tutorials/job-metrics/tutorial.md]: Seconds looks nicer to me too.

[the remark this answers is not in the document]
Comment (natefoo) [reply topics/admin/tutorials/job-metrics/tutorial.md]: I think it's also worth discussing that participants don't understand where variables come from. We do try to explain this but it's confusing and quickly forgotten I think.

1. From role defaults/vars or `set_fact`s
2. From facts
3. From other vars we define
4 ...?

[the remark this answers is not in the document]
Comment (natefoo) [reply topics/admin/tutorials/job-metrics/tutorial.md]: FWIW my vote is for the second one as well.

[the remark this answers is not in the document]
Comment (hexylena) [reply topics/admin/tutorials/job-metrics/tutorial.md:99]: OK, second one it is.

**Comment by hexylena** [comment]
Ready for review if anyone has time today @galaxyproject/training-admin

**Comment by hexylena** [comment]
@TKlingstrom hey since you're working on admin stuff, let me know if you have any comments on this?

**Comment by shiltemann** [comment]
thanks @hexylena! and thanks for the review @natefoo @Slugger70 @nsoranzo and @TKlingstrom!

File: topics/admin/tutorials/connect-to-compute-cluster/slides.html [hunk 25]
```diff
@@ -296,128 +296,7 @@
 
 ---
 
-# Galaxy Dependency Resolver Configuration
```
Comment (nsoranzo) [inline topics/admin/tutorials/connect-to-compute-cluster/slides.html:299]: @hexylena The text removed here contained more (I think useful) material than what's in `topics/admin/tutorials/toolshed/slides.html` . Also I've been thinking what's the more appropriate slide deck to include such info about dependency resolution, and I feel it should be in `topics/admin/tutorials/tool-management/slides.html` (there was one slide, which I removed in d5ba4aa1f5d8c6e2ab3ef4e63897478702a958c6 :blush:).

What do you think? I can do the work if you agree.

**Comment by hexylena** [comment]
I think that sounds great, go for it! Thanks for looking at it

fre. 29. mai 2020, 14:59 skrev Nicola Soranzo <notifications@github.com>:

**Pull request merged**